src/MongoDB/Cursor.c

```diff
@@ -366,9 +309,72 @@ static PHP_METHOD(Cursor, getIterator)
 	}
 	zend_restore_error_handling(&error_handling);
 
-	zend_create_internal_iterator_zval(return_value, getThis());
+	data = php_phongo_cursor_get_current_data(intern);
+
+	if (data) {
+		ZVAL_COPY_DEREF(return_value, data);
+	}
+}
+
+PHP_METHOD(Cursor, key)
+{
+	zend_error_handling  error_handling;
+	php_phongo_cursor_t* intern = Z_CURSOR_OBJ_P(getThis());
+
+	zend_replace_error_handling(EH_THROW, phongo_exception_from_phongo_domain(PHONGO_ERROR_INVALID_ARGUMENT), &error_handling);
+	if (zend_parse_parameters_none() == FAILURE) {
+		zend_restore_error_handling(&error_handling);
+		return;
+	}
+	zend_restore_error_handling(&error_handling);
+
+	php_phongo_cursor_get_current_key(intern, return_value);
+}
+
+PHP_METHOD(Cursor, next)
+{
+	zend_error_handling  error_handling;
+	php_phongo_cursor_t* intern = Z_CURSOR_OBJ_P(getThis());
+
+	zend_replace_error_handling(EH_THROW, phongo_exception_from_phongo_domain(PHONGO_ERROR_INVALID_ARGUMENT), &error_handling);
+	if (zend_parse_parameters_none() == FAILURE) {
+		zend_restore_error_handling(&error_handling);
+		return;
+	}
+	zend_restore_error_handling(&error_handling);
+
+	php_phongo_cursor_move_forward(intern);
+}
```

Comment:
next() should return the value as well.

    php_phongo_cursor_move_forward(intern);
    zval* data = php_phongo_cursor_get_current_data(intern);
    if (data) {
        ZVAL_COPY_DEREF(return_value, data);
    }

Reply:
Ignore this comment. I was wrong about it.